### PR TITLE
Document Hive text write serialization format checks

### PR DESCRIPTION
**Pull request by mythrocks**
This commit documents the serialization format checks for writing Hive text, and why it differs from the read-side.

`spark-rapids` supports only '^A'-separated Hive text files for read and write. This format tends to be denoted in a Hive table's Storage Properties with `serialization.format=1`.

If a Hive table is written with a different/custom delimiter, it is denoted with a different value of `serialization.format`.  For instance, a CSV table might be denoted by `serialization.format='', field.delim=','`.

It was noticed in https://github.com/NVIDIA/spark-rapids/issues/11803 that:
1. On the [read side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala#L155-L161), `spark-rapids` treats an empty `serialization.format` as `''`.
2. On the [write side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveFileFormat.scala#L130-L136), an empty `serialization.format` is seen as `1`.

The reason for the read side value is to be conservative.  Since the table is pre-existing, its value should have been set already.

The reason for the write side is that there are legitimate cases where a table might not have its `serialization.format` set.  (CTAS, for one.)

This commit documents all the scenarios that need to be considered on the write side.

### CODE REVIEW

**Review by jihoonson — COMMENTED** [review]
This document looks good and helpful. I wonder though, should we document this for our users as well?

**Comment by mythrocks** [comment]
Build

**Comment by jihoonson** [comment]
> 1. On the [read side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala#L155-L161), spark-rapids treats an empty serialization.format as ''.
> 2. On the [write side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveFileFormat.scala#L130-L136), an empty serialization.format is seen as 1.

@mythrocks oh I was wondering mostly about the part quoted above as well as the fact that the plugin only supports one particular serialization format. As for the latter, I think we should document it to clarify the scope of what is supported today. I'm not sure though if it's already documented somewhere. As for the former, it seems reasonable to me to document them as well so that the user could know what is expected and may help their troubleshooting.

**Comment by mythrocks** [comment]
> > 1. On the [read side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala#L155-L161), spark-rapids treats an empty serialization.format as ''.
> > 2. On the [write side](https://github.com/NVIDIA/spark-rapids/blob/aa2da410511d8a737e207257769ec662a79174fe/sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveFileFormat.scala#L130-L136), an empty serialization.format is seen as 1.
> 
> @mythrocks oh I was wondering mostly about the part quoted above as well as the fact that the plugin only supports one particular serialization format. 

I might have again done a poor job of explaining this.  Let me try again. 

The *only* case where the user sees `serialization.format=""` is if they used a customized format, unsupported in `spark-rapids` for both write and read.  They should already know (from documentation and exception messages) that we don't support this.

CTAS is a special case where `spark-rapids` will see the table's definition *before* it has been constructed.  By the time the user sees the table, the construction would have been complete, and `serialization.format=1`.  But before construction, `spark-rapids` would see it as unset.  

It is CTAS's transient state that I'm trying to document.  (We will see it as unset, and have to treat that as `^A`-separated.).  The transient state is not visible to the user, and does not need to be documented for them.

**Comment by jihoonson** [comment]
Ah I see, it makes sense now. Thanks for the explanation.

**Review by jihoonson — APPROVED** [review]

**Comment by mythrocks** [comment]
@jihoonson, thank you for your review and the subsequent discussion.  I've merged this change.
